docs/reference/api/storage/SlicingBlockDevice.md

```diff
@@ -1,11 +1,19 @@
 ## SlicingBlockDevice
 
-[Add description here.]
+The SlicingBlockDevice class provides a way to break up a block device into subunits without the need to manually track offsets. The SlicingBlockDevice acts as an opposite of the [ChainingBlockDevice](/docs/v5.6/reference/chainingblockdevice.html) class.
+
+The constructor takes in the master block device pointer and the start and end addresses of where you would like to partition the sub-block. By not specifying the end address, you create a block device that spans from the provided start address to the end of the underlying block device.
+
+  - _bd_ -  Block device to back the SlicingBlockDevice.
+  - _start_ - Start block address to map to block 0. Negative addresses are calculated from the end of the underlying block device.
+  - _end_ - End block address to mark the end of the block device. This block is not mapped; negative addresses are calculated from the end of the underlying block device.
 
 ### SlicingBlockDevice class reference
 
 [![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/v5.6/mbed-os-api-doxy/class_slicing_block_device.html)
 
 ### SlicingBlockDevice example
 
-[Add example here.]
+This SlicingBlockDevice example partitions a [HeapBlockDevice](/docs/v5.6/reference/heapblockdevice.html) into three subunits and showcases programming and reading back data segments through both the underlying master block device and the sliced subunits.
```

Comment:
not sure what the policy is on links to specific versions but in general I'm not a fan of linking but happy to be told otherwise.

Reply:
Fair enough, I could pull the links out. I don't feel particularly strongly one way or the other.
@AnotherButler Would you rather have the links or no?

Reply:
@AnotherButler The specific version issue should be worked around by your new release update script right?  Although we should probably be consistent on when we're linking and when we're not. If we reference every external class with a link I could see that getting out of hand.